crates/integration/src/tests.rs

```diff
@@ -234,6 +234,33 @@ fn signed_remainder() {
     run_differential(actions);
 }
 
+#[test]
+fn balance_of() {
+    let mut actions = instantiate("contracts/Value.sol", "Value");
+
+    actions.push(Call {
+        origin: TestAddress::Alice,
+        dest: TestAddress::Instantiated(0),
+        value: 0,
+        gas_limit: None,
+        storage_deposit_limit: None,
+        data: Contract::value_balance_of(Address::from(CHARLIE.to_fixed_bytes())).calldata,
+    });
+
+    actions.push(VerifyCall(VerifyCallExpectation {
+        success: true,
+        // Charlie's balance without ExistentialDeposit
+        output: OptionalHex::from(U256::from(1_000_000_000 - 1_000).to_be_bytes_vec()),
+        gas_consumed: None,
+    }));
+
+    Specs {
+        actions,
+        ..Default::default()
+    }
+    .run();
+}
+
 /*
 // These test were implement for the mock-runtime and need to be ported yet.
```

Comment:
This is actually one of those yet-to-be ported :)

To easily test this against EVM, the `Value.sol` integration test just needs a slight change. Basically because we aren't able to express using the address of an instantiated account as calldata with the current spec data. Just move the assertions into Solidity:

```solidity
// SPDX-License-Identifier: MIT
pragma solidity ^0.8;

/* runner.json
{
    "differential": true,
    "actions": [
        {
            "Upload": {
                "code": {
                    "Solidity": {
                        "contract": "ValueTester"
                    }
                }
            }
        },
        {
            "Instantiate": {
                "value": 1024,
                "code": {
                    "Solidity": {
                        "contract": "Value"
                    }
                }
            }
        },
        {
            "Call": {
                "dest": {
                    "Instantiated": 0
                },
                "value": 123,
                "data": "3fa4f245"
            }
        }
    ]
}
*/

contract ValueTester {
    constructor() payable {}

    function balance_self() public view returns (uint ret) {
        ret = address(this).balance;
    }
}

contract Value {
    constructor() payable {
        ValueTester tester = new ValueTester{value: msg.value}();

        // own account
        assert(address(this).balance == 0);

        // tester account
        assert(address(tester).balance == msg.value);
        assert(tester.balance_self() == msg.value);

        // non-existant account
        assert(address(0xdeadbeef).balance == 0);
    }

    function value() public payable returns (uint ret) {
        ret = msg.value;
    }
}
```

Reply:
A nice side effect of doing the assertions in Solidity is that we don't even have to consider things like the ED. We just expect the contract to observe the exact same values than what it would see in the EVM reference implementation :)

Reply:
That's a great idea. Will update